doc/whats_new/v0.20.rst

```diff
@@ -67,6 +67,13 @@ Changelog
   those estimators as part of parallel parameter search or cross-validation.
   :issue:`12122` by :user:`Olivier Grisel <ogrisel>`.
 
+:mod:`sklearn.metrics`
+........................
+
+- |Fix| Fixed a bug in :func:`metrics.silhouette_score` where diagonal entries
+  of a precomputed distance matrix affected the final score.
+  :issue:`12258` by :user:`Stephen Tierney`.
```

Comment:
Yes, core devs get a simplified syntax. This syntax, provided by the sphinx-issues plugin, generates a link to GitHub. But you need something like ```:user:`Stephen Tierney <sjtrny>` ```